### PR TITLE
nest: fix Linux build and macOS ncurses detection

### PR DESCRIPTION
**Pull request by cho-m**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/nest.rb [hunk 1]
```diff
@@ -16,35 +16,43 @@ class Nest < Formula
...
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args, *args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+
     # Replace internally accessible gcc with externally accessible version
     # in nest-config if required
-    inreplace bin/"nest-config",
-        %r{#{HOMEBREW_REPOSITORY}/Library/Homebrew/shims.*/super}o,
-        "#{HOMEBREW_PREFIX}/bin"
+    inreplace bin/"nest-config", Superenv.shims_path/ENV.cxx, ENV.cxx
```
Comment (cho-m) [inline Formula/nest.rb:55]: Previously was setting `$HOMEBREW_PREFIX/bin/clang++`, which wouldn't exist unless user installed `llvm` and then manually linked it as keg_only.

File: Formula/nest.rb [hunk 2]
```diff
@@ -16,35 +16,43 @@ class Nest < Formula
...
+      -Dwith-openmp=ON
+    ]
+    if OS.mac?
+      libomp = Formula["libomp"]
+      args << "-Dwith-libraries=#{libomp.opt_lib}/libomp.dylib"
+      args << "-DOpenMP_C_FLAGS=-Xpreprocessor\ -fopenmp\ -I#{libomp.opt_include}"
+      args << "-DOpenMP_C_LIB_NAMES=omp"
+      args << "-DOpenMP_CXX_FLAGS=-Xpreprocessor\ -fopenmp\ -I#{libomp.opt_include}"
+      args << "-DOpenMP_CXX_LIB_NAMES=omp"
+      args << "-DOpenMP_omp_LIBRARY=#{libomp.opt_lib}/libomp.dylib"
```
[the remark this answers is not in the document]
Comment (cho-m) [reply Formula/nest.rb:42]: It is able to detect `libomp`, at least for Intel build, so can try removing:
```
-- Found Readline: /usr/local/opt/readline/lib/libreadline.dylib (found version "8.1")
-- Found PkgConfig: /usr/local/bin/pkg-config (found version "0.29.2")
-- Found GSL: /usr/local/include (found version "2.7.1")
-- Found OpenMP_C: -Xclang -fopenmp (found version "5.0")
-- Found OpenMP_CXX: -Xclang -fopenmp (found version "5.0")
-- Found OpenMP: TRUE (found version "5.0")
```

```
//CXX compiler flags for OpenMP parallelization
OpenMP_CXX_FLAGS:STRING=-Xclang -fopenmp

//CXX compiler libraries for OpenMP parallelization
OpenMP_CXX_LIB_NAMES:STRING=libomp

//C compiler flags for OpenMP parallelization
OpenMP_C_FLAGS:STRING=-Xclang -fopenmp

//C compiler libraries for OpenMP parallelization
OpenMP_C_LIB_NAMES:STRING=libomp

//Path to a library.
OpenMP_libomp_LIBRARY:FILEPATH=/usr/local/lib/libomp.dylib
```

[the remark this answers is not in the document]
Comment (Bo98) [reply Formula/nest.rb]: Yep the issue before was `-Xclang` was not supported by our shims but `-Xpreprocessor` was - both are now.

File: Formula/nest.rb [hunk 3]
```diff
@@ -16,35 +16,33 @@ class Nest < Formula
...
-    args << "-Dwith-openmp=ON"
-    args << "-Dwith-libraries=#{libomp.opt_lib}/libomp.dylib"
-    args << "-DOpenMP_C_FLAGS=-Xpreprocessor\ -fopenmp\ -I#{libomp.opt_include}"
-    args << "-DOpenMP_C_LIB_NAMES=omp"
-    args << "-DOpenMP_CXX_FLAGS=-Xpreprocessor\ -fopenmp\ -I#{libomp.opt_include}"
-    args << "-DOpenMP_CXX_LIB_NAMES=omp"
-    args << "-DOpenMP_omp_LIBRARY=#{libomp.opt_lib}/libomp.dylib"
+  def install
+    args = %W[
+      -DCMAKE_INSTALL_PREFIX:PATH=#{prefix}
```
Comment (Bo98) [inline Formula/nest.rb:32]:
```suggestion
```

This is in `std_cmake_args`.

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**